### PR TITLE
Pe 39749 gh deployments

### PR DESCRIPTION
**Pull request by khaledm1990**
This PR is for to integration GH  Deployment  with Jira 
check [here](https://github.com/freeletics/fl-bolt/pull/292) for more details

### CODE REVIEW

File: .github/workflows/gha-deploy-integration.yaml [hunk 1]
```diff
@@ -82,24 +83,28 @@ jobs:
...
-
-      - uses: int128/kaniko-action@v1.43.0
-        name: Build with Kaniko
-        with:
-          push: true
-          file: "${{ inputs.DOCKERFILE }}"
-          tags: ${{ steps.metadata.outputs.tags }}
-          labels: ${{ steps.metadata.outputs.labels }}
-          cache: ${{ inputs.KANIKO_CACHE }}
-          cache-repository: ${{ inputs.AWS_ACCOUNT_ID }}.dkr.ecr.${{ inputs.AWS_REGION }}.amazonaws.com/kaniko-cache
```
Comment (gabrielittner) [inline .github/workflows/gha-deploy-integration.yaml:94]: Isn't this still needed to actually build the image?

File: .github/workflows/gha-create-tag.yaml [hunk 2]
```diff
@@ -47,7 +48,23 @@ jobs:
               ref: "refs/tags/deploy-prod-${{ steps.date.outputs.date }}",
               sha: context.sha
             })
-      # adding this here so if the repo doe not have the release workflow it will update here
+      - uses: chrnorm/deployment-action@v2
+        name: Create GitHub deployment
+        id: deployment
+        with:
+          token: '${{ github.token }}'
+          environment: integration
```
Comment (gabrielittner) [inline .github/workflows/gha-create-tag.yaml:56]:
```suggestion
          environment: production
```

File: .github/workflows/gha-auto-deploy-integration.yaml [hunk 3]
```diff
@@ -93,13 +94,18 @@ jobs:
...
+          environment: integration
+          initial-status: in_progress
+          payload: "{\"docker_image_tag\":\"gha-deploy-int-${{ steps.timestamp.outputs.date }}\"}"
+      - name: Update deployment status (in_progress)
+        uses: chrnorm/deployment-status@v2
+        with:
+          token: '${{ github.token }}'
+          environment-url: http://my-app-url.com
+          state: 'in_progress'
+          deployment-id: ${{ steps.deployment.outputs.deployment_id }}
```
[the remark this answers is not in the document]
Comment (khaledm1990) [reply .github/workflows/gha-auto-deploy-integration.yaml:111]: Not sure but based on the did not try it but based based on the Jira doc

> 
The GitHub for Jira app only listens to deployment_status events. Just creating a deployment via the [create deployment API](https://docs.github.com/en/rest/reference/repos#create-a-deployment) (or the chrnorm/deployment-action@releases/v1 action) isn't enough. After creating a deployment you need to call the [create deployment status API](https://docs.github.com/en/rest/reference/repos#create-a-deployment-status) (or the chrnorm/deployment-status@releases/v1 action) at least once to provide a status for that deployment (success, failure, pending, ...).

File: .github/workflows/gha-deploy-integration.yaml (hunk 1 above)
Comment (khaledm1990) [reply .github/workflows/gha-deploy-integration.yaml:94]: My bad removed by mistake

File: .github/workflows/gha-auto-deploy-integration.yaml [hunk 4]
```diff
@@ -93,13 +94,18 @@ jobs:
           cache: ${{ inputs.KANIKO_CACHE }}
           cache-repository: ${{ inputs.AWS_ACCOUNT_ID }}.dkr.ecr.${{ inputs.AWS_REGION }}.amazonaws.com/kaniko-cache
 
-  jira_markers:
```
Comment (khaledm1990) [inline .github/workflows/gha-auto-deploy-integration.yaml:96]: It was not used, we do not need any more

File: .github/workflows/web.yaml [hunk 5]
```diff
@@ -11,6 +11,11 @@ on:
         required: false
         type: string
         default: "Dockerfile"
+      deployment_type:
```
Comment (khaledm1990) [inline .github/workflows/web.yaml:14]: Added this to differentiate between an integration which was triggered manually or automatically by merging a branch to master

**Review by gabrielittner — APPROVED** [review]

**Pull request merged**